shpc/main/client.py

```diff
@@ -124,13 +124,12 @@ def update(self, name=None, dryrun=False, filters=None):
         """
         # No name provided == "update all"
         if name:
-            modules = [name]
-        else:
-            modules = [x[1] for x in list(self.registry.iter_modules())]
-
-        for module_name in modules:
-            config = self._load_container(module_name)
+            config = self._load_container(name)
```

Comment:
I think generally it's good to be consistent with respect to how we handle the list vs. all - e.g., we prepare a list of names for both cases and then call the same functions. With the changes here, it's technically done two different ways, and that could eventually lead to bugs.

Reply:
I improved it the best I can. Otherwise, we need to:

- either inline `_load_container` and `load_registry_config`, duplicating their logic
- or revert to the original inefficient search (ask all registries for their module names and then, for each module name, ask each registry if it knows about it)